repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/scripting/ScriptingLanguageTest.groovy

```diff
@@ -0,0 +1,116 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package features.filters.scripting
+
+import framework.ReposeConfigurationProvider
+import framework.ReposeLogSearch
+import framework.ReposeValveLauncher
+import framework.TestProperties
+import org.rackspace.deproxy.Deproxy
+import org.rackspace.deproxy.MessageChain
+import org.rackspace.deproxy.Response
+import spock.lang.Specification
+import spock.lang.Unroll
+
+/**
+ * Created by jennyvo on 4/1/16.
+ */
+class ScriptingLanguageTest extends Specification {
+
+    int reposePort
+    int targetPort
+    String url
+    TestProperties properties
+    ReposeConfigurationProvider reposeConfigProvider
+    ReposeLogSearch reposeLogSearch
+    ReposeValveLauncher repose
+    Map params = [:]
+    Deproxy deproxy
+
+    def setup() {
+        properties = new TestProperties()
+        this.reposePort = properties.reposePort
+        this.targetPort = properties.targetPort
+        this.url = properties.reposeEndpoint
+
+        params = properties.getDefaultTemplateParams()
+
+        // start a deproxy
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(this.targetPort)
+
+        // setup config provider
+        reposeConfigProvider = new ReposeConfigurationProvider(properties.getConfigDirectory(), properties.getConfigTemplates())
+
+    }
+
+    def cleanup() {
+        if (repose) {
+            repose.stop()
+        }
+        if (deproxy) {
+            deproxy.shutdown()
+        }
+    }
+
+    @Unroll ("Test with support language: #language")
+    def "Test with all support languages scripting"() {
+        given:
+        reposeConfigProvider.cleanConfigDirectory()
+        reposeConfigProvider.applyConfigs("common", params)
+        reposeConfigProvider.applyConfigs("features/filters/scripting", params)
+        reposeConfigProvider.applyConfigs("features/filters/scripting/" + language, params)
+
+        // start repose
+        repose = new ReposeValveLauncher(
+                reposeConfigProvider,
+                properties.getReposeJar(),
+                url,
+                properties.getConfigDirectory(),
+                reposePort
+        )
+        repose.enableDebug()
+        reposeLogSearch = new ReposeLogSearch(properties.getLogFile());
+        repose.start(killOthersBeforeStarting: false,
```

Comment:
We could stop it in the method, but we won't be left with 4 instances running after this test in this case. The `cleanup` method will handle stopping Repose.

Side note: Deproxy should be shutdown in the `cleanupSpec` method, not the `cleanup` method.